### PR TITLE
feat(frontend): added fix for the base url

### PR DESCRIPTION
**Pull request by devamitranjan**
# Motivation

Added fix for the buildURL() when API url has a path

Fixes https://github.com/Hexastack/Hexabot/issues/114

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (Storybook)
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/utils/URL.ts [hunk 1]
```diff
@@ -28,7 +28,11 @@ export const getFromQuery = ({
 
 export const buildURL = (baseUrl: string, relativePath: string): string => {
```
Comment (marrouchi) [inline frontend/src/utils/URL.ts:29]: Thank you @devamitranjan 👏  can you please copy over the same function to the api side : api/src/utils/helpers/URL.ts

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: frontend/src/utils/URL.ts [hunk 2]
```diff
@@ -28,7 +28,11 @@ export const getFromQuery = ({
...
   try {
-    const url = new URL(relativePath, baseUrl);
+    const normalizedBaseUrl = baseUrl.endsWith("/") ? baseUrl : `${baseUrl}/`;
```
Comment (yassinedorbozgithub) [inline frontend/src/utils/URL.ts:31]: Hi @devamitranjan, i can notice that this logic will expect to have the relativePath prop to be relative, what if we receive an absolute one.
`buildURL('http://my-domain.com/api/','http://github.com/'); `
The output will be => http://github.com/

File: api/src/utils/helpers/URL.ts [hunk 3]
```diff
@@ -1,6 +1,10 @@
 export const buildURL = (baseUrl: string, relativePath: string): string => {
   try {
-    const url = new URL(relativePath, baseUrl);
+    const normalizedBaseUrl = baseUrl.endsWith('/') ? baseUrl : `${baseUrl}/`;
```
Comment (yassinedorbozgithub) [inline api/src/utils/helpers/URL.ts:3]: Hi @devamitranjan, i can notice that this logic will expect to have the relativePath prop to be relative, what if we receive an absolute one.
`buildURL('http://my-domain.com/api/','http://github.com/'); `
The output will be => http://github.com/

Comment (devamitranjan) [reply api/src/utils/helpers/URL.ts]: Okay @yassinedorbozgithub added the change.

File: frontend/src/utils/URL.ts (hunk 2 above)
Comment (devamitranjan) [reply frontend/src/utils/URL.ts]: Okay @yassinedorbozgithub added the change.

**Comment by devamitranjan** [comment]
@marrouchi can we merge this if all okay?

**Review by marrouchi — APPROVED** [review]
Yup, it looks solid 🥇 thanks !

**Pull request merged**